### PR TITLE
oAuth input parameters UI

### PR DESCRIPTION
**Pull request by avida**
## What
UI part for #6971
All parameters list on oauthFlowInputFields are validated before sending get_consent_url reque
Here is how validation looks like: 

Peek 2021-10-25 12-04

## How
*Describe the solution*

## Recommended reading order
1. `x.java`
2. `y.python`

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector added to connector index like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

**Review by jamakase — CHANGES_REQUESTED** [review]
Awesome work @avida, but I have to reject such a PR as it requires significant changes for now and doesn't cover all UI cases.

First of all, useConnectorAuth is a service that is responsible only for auth aspect. Not for setting validation errors and etc. This part should be handled as separate flow if we are going to set up any auth errors.

Secondly, it doesn't cover all UI cases - check slack discussion where I provided detailed info, why it won't work.
- What happens if user decides to update one of dependent fields and oauth was not run again?
- What if user passes auth for one case but then change some of the fields?
- What if user tried to submit request, got errors in some fields and then tried to pass auth? It should be obvious which fields are required for auth and which not.
- What if form is long enough - some error should be displayed within oauthFlow ( above or below the button ) that some fields are required, so user will know why his flow is not starting.

What is more, I and @jrhizor confirmed that we will go next path:
```
We could build more of a wizard view based on the dependencies (would likely have more overlap with the future approach) purely on the frontend side
```
As it is easier for future refactoring.

Though if this PR is fixed properly, where we could easily adjust for future needs - I would be glad to accept it as temporary fix.

**Review by jamakase — COMMENTED** [review]
UI still doesn't cover other cases that actually should be covered and function still handles too much logic and should be split.

File: airbyte-webapp/src/hooks/services/useConnectorAuth.tsx [hunk 1]
```diff
@@ -105,18 +111,46 @@ export function useRunOauthFlow(
...
+  const {
+    values,
+    setValues,
+    validateForm,
+    setFieldTouched,
+  } = useFormikContext();
+  const oAuthFieldsToValidate = useMemo(() => {
+    const oAuthRequired =
+      connector?.authSpecification?.oauth2Specification?.oauthFlowInputFields;
+    return oAuthRequired ? oAuthRequired.map((value) => value.join(".")) : [];
```
Comment (timroes) [inline airbyte-webapp/src/hooks/services/useConnectorAuth.tsx:123]: :art: Suggestions for a bit better code style:

```suggestion
    return oAuthRequired?.map((value) => value.join(".")) ?? [];
```

File: airbyte-webapp/src/hooks/services/useConnectorAuth.tsx [hunk 2]
```diff
@@ -105,18 +111,46 @@ export function useRunOauthFlow(
...
   const [{ loading }, onStartOauth] = useAsyncFn(async () => {
-    const consentRequestInProgress = await getConsentUrl(connector);
-
-    param.current = consentRequestInProgress.payload;
-    openWindow(consentRequestInProgress.consentUrl);
-  }, [connector]);
+    const errors = (await validateForm(values)) as ServiceFormValues;
+    const oAuthErrors = Object.keys(
+      flatten(errors.connectionConfiguration)
+    ).filter((path) => oAuthFieldsToValidate.indexOf(path) != -1);
```
Comment (timroes) [inline airbyte-webapp/src/hooks/services/useConnectorAuth.tsx:138]: :art: Suggestions for a bit better code style:

```suggestion
    ).filter((path) => oAuthFieldsToValidate.includes(path));
```

File: airbyte-webapp/src/hooks/services/useConnectorAuth.tsx (hunk 1 above)
Comment (Jamakase) [reply airbyte-webapp/src/hooks/services/useConnectorAuth.tsx:123]: Thanks @timroes, but I have already applied same changes in separate branch that will be merged into this one. No need to make same changes twice :)

File: airbyte-webapp/src/hooks/services/useConnectorAuth.tsx (hunk 2 above)
Comment (Jamakase) [reply airbyte-webapp/src/hooks/services/useConnectorAuth.tsx:138]: Actually I am going to experiment with setTouched and validate: true. Then we won't need to check for includes. 

But thanks for your proposal!

**Review by jamakase — APPROVED** [review]

**Pull request merged**